scripts/mypy-failing.txt

```diff
@@ -25,6 +25,7 @@ pytensor/tensor/random/basic.py
 pytensor/tensor/random/op.py
 pytensor/tensor/random/utils.py
 pytensor/tensor/rewriting/basic.py
+pytensor/tensor/shape.py
```

Comment:
This shouldn't be done. Over time we want less files, not more. What's mypy complaining about?

Reply:
Mypy complains about the line ```from numpy.core.numeric import normalize_axis_tuple```. The message is that ```numpy.core.numeric``` does not have ```normalize_axis_tuple```.
This already occurs in other places as well and are ignored in the mypy-failing-list. For eg:
https://github.com/pymc-devs/pytensor/blob/5e612abbc8fe11d9f1a723154521c0a68070c1a9/pytensor/tensor/basic.py#L18

Reply:
You can do the same. Better to ignore a single line than a whole file.

In numpy 2.0 at least they moved this to a user facing location so it should be legal to import then. I don't know if it's already there in the current releases.

Reply:
It already exists. Can you check if it's fine to import from here:
 https://numpy.org/devdocs/reference/generated/numpy.lib.array_utils.normalize_axis_tuple.html#:~:text=Normalizes%20an%20axis%20argument%20into,from%20being%20specified%20multiple%20times.

If so we should import like this in the other places as remove the type ignore

Reply:
I'll just try this out 👍

Reply:
Also https://numpy.org/devdocs/reference/generated/numpy.lib.array_utils.normalize_axis_index.html#numpy.lib.array_utils.normalize_axis_index

Reply:
So I checked I am currently using numpy version 1.26.4 and I get the error ```AttributeError: module 'numpy.lib' has no attribute 'array_utils'```. I checked the site-packages and it seems like this feature is introduced in numpy 2.0 and greater. Here's the reference ```numpy.lib``` ```_init_.py``` that matches with what I have locally:

https://github.com/numpy/numpy/blob/ee81e365a87963b695b688d1b38a357599fa61db/numpy/lib/__init__.py#L1-L92

We are yet to switch to numpy2.0 right?

Reply:
Strange because the docs say it's new from v1.13.

Anyway that answers it for now. No we're not switching to numpy 2.0 for a while, so let's keep importing and telling mypy to ignore it